resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -75,7 +79,8 @@ object DbReader {
               WHERE
                 c2.cloudContext = c1.cloudContext AND
                 c2.runtimeName = c1.runtimeName AND
-                (c2.status = "Stopped" OR c2.status = "Running")
+                c2.status != "Deleted" AND
+                c2.status != "Deleting"
```

Comment:
two questions:
- do we need to or want to modify any of the queries below?, and
- this looks like we are saying we want `status = deleting` matching the `dateAccessed` criteria above and excluding all non-`Deleted and Deleting` environments (which I think is another way of saying return all runtimes matching deleting and the date config) - which makes me want to ask, do we need this sub-query?

Reply:
>  do we need to or want to modify any of the queries below?
It might make sense to apply the same thing for `erroredRuntimeQuery` and `stoppedRuntimeQuery`...but I'd like to limit this PR to just what's necessary for now in case of any surprises.

> this looks like we are saying we want status = deleting matching the dateAccessed criteria above and excluding all non-Deleted and Deleting environments (which I think is another way of saying return all runtimes matching deleting and the date config) - which makes me want to ask, do we need this sub-query?
It's needed becuz AoU reuses runtime name for same project and user. See the example I listed in the description. Basically, there will be many "Deleted" or "Deleting" AoU runtimes with same project and name, and if there's an "active" runtime in the same project and name, then we want to skip these Deleted/Deleting runtimes, because if we don't skip, then we'll end up doing sth like `gcp.deleteRuntime(project, name)`, and it'll delete the active one while it shouldn't be deleted

Reply:
- 👍 on the first question, thank you. (makes sense), and
- makes sense re: name conflicts as you describe. do you know if because we are doing a `select *` in the `NOT EXISTS`-query and are including the `cluster.id in the global `SELECT ...`-statement above if we will actually be filtering out based on the cluster `name`? Asked another way, do we need to explicitly say `where name not in (insert-subquery-here)`?

Reply:
>Asked another way, do we need to explicitly say where name not in (insert-subquery-here)`?
I don't think we need to...as long as `NOT EXISTS` returns anything, we'll exclude that runtime. So basically we're saying, for every `Deleting`/`Deleted` runtime, does there exist another one with the same name and same project, and its status is "active". If there is any such row in returned in the subquery, we'll exclude the runtime

Reply:
oh ... shoot ... my bad. I missed the `c1` amidst all the `c2`'s in the subquery. thanks for entertaining my questions!